esp-hal/src/dma/mod.rs

```diff
@@ -1285,15 +1294,77 @@ pub trait ChannelTypes {
     type P: PeripheralMarker;
     type Tx<'a>: Tx;
     type Rx<'a>: Rx;
+    type Binder: InterruptBinder;
+}
+
+pub trait InterruptBinder {
```

Comment:
We also need to cfg away the trait method here for `cfg(feature = "vectored")`? The impl cfg's it away below.